src/kaleidoscope/Kaleidoscope.h

```diff
@@ -132,6 +132,28 @@ class Kaleidoscope_ {
     return millis_at_cycle_start_;
   }
 
+  /** Determines if a timer has expired.
+   *
+   * This method should be used whenever checking to see if a timeout has been
+   * reached. It correctly computes timeout values even when integer overflow
+   * occurs, given a start time and a timeout. It takes two parameters:
+   *
+   * - start_time: A timestamp when the timer started, which should be set by
+   *      calling `Kaleidoscope.millisAtCycleStart()`. It can be any integer
+   *      type.
+   *
+   * - ttl: The timeout value or interval to check (ttl = "time to live"). The
+   *      timer expires (and `hasTimeExpired()` returns true) when the time that
+   *      has elapsed since `start_time` exceeds this value. It must be and
+   *      integer type that is no bigger than the type of `start_time`.
+   */
+  template <typename _Timestamp = uint16_t, typename _Timeout = uint16_t>
```

Comment:
As @noseglasses pointed out in #642, these default values are mostly pointless, except perhaps as a hint for proper usage.